[xpath] Modernize /fast/xpath tests

zakariaridouh opened the pull request:
#### 8e00808b647919e9d256ff425f6fdd47085a6e6e
<pre>
[xpath] Modernize /fast/xpath tests
<a href="https://bugs.webkit.org/show_bug.cgi?id=289648">https://bugs.webkit.org/show_bug.cgi?id=289648</a>
<a href="https://rdar.apple.com/146892392">rdar://146892392</a>

Reviewed by Tim Nguyen.

Modernize the tests in `fast/xpath` to use `test-js.js` and drop the
console element.

* LayoutTests/fast/xpath/4XPath/Borrowed/cz_20030217.html:
* LayoutTests/fast/xpath/4XPath/Borrowed/kd_20010423.html:
* LayoutTests/fast/xpath/4XPath/Borrowed/od_20000608.html:
* LayoutTests/fast/xpath/4XPath/Borrowed/rs_20010831.html:
* LayoutTests/fast/xpath/4XPath/Borrowed/sr_20021217.html:
* LayoutTests/fast/xpath/4XPath/Core/test_boolean_expr.html:
* LayoutTests/fast/xpath/4XPath/Core/test_core_functions-expected.txt:
* LayoutTests/fast/xpath/4XPath/Core/test_core_functions.html:
* LayoutTests/fast/xpath/4XPath/Core/test_literal_expr.html:
* LayoutTests/fast/xpath/4XPath/Core/test_location_path.html:
* LayoutTests/fast/xpath/4XPath/Core/test_node_test.html:
* LayoutTests/fast/xpath/4XPath/Core/test_nodeset_expr.html:
* LayoutTests/fast/xpath/4XPath/Core/test_numeric_expr.html:
* LayoutTests/fast/xpath/4XPath/Core/test_parser.html:
* LayoutTests/fast/xpath/4XPath/Core/test_predicate_list.html:
* LayoutTests/fast/xpath/4XPath/Core/test_step.html:
* LayoutTests/fast/xpath/ambiguous-operators-expected.txt:
* LayoutTests/fast/xpath/ambiguous-operators.html:
* LayoutTests/fast/xpath/attr-namespace.html:
* LayoutTests/fast/xpath/attribute-node-predicate.html:
* LayoutTests/fast/xpath/detached-subtree-invalidate-iterator.html:
* LayoutTests/fast/xpath/document-order.html:
* LayoutTests/fast/xpath/evaluate-twice.html:
* LayoutTests/fast/xpath/invalid-functions.html:
* LayoutTests/fast/xpath/node-name-case-sensitivity-expected.txt:
* LayoutTests/fast/xpath/node-name-case-sensitivity.html:
* LayoutTests/fast/xpath/null-namespace-in-html-expected.txt:
* LayoutTests/fast/xpath/null-namespace-in-html.html:
* LayoutTests/fast/xpath/position.html:
* LayoutTests/fast/xpath/py-dom-xpath/abbreviations.html:
* LayoutTests/fast/xpath/py-dom-xpath/data.html:
* LayoutTests/fast/xpath/py-dom-xpath/expressions.html:
* LayoutTests/fast/xpath/py-dom-xpath/functions-expected.txt:
* LayoutTests/fast/xpath/py-dom-xpath/functions.html:
* LayoutTests/fast/xpath/py-dom-xpath/nodetests.html:
* LayoutTests/fast/xpath/py-dom-xpath/paths.html:
* LayoutTests/fast/xpath/py-dom-xpath/predicates.html:
* LayoutTests/fast/xpath/reverse-axes.html:
* LayoutTests/fast/xpath/string-value.html:
* LayoutTests/fast/xpath/substring-after-expected.txt:
* LayoutTests/fast/xpath/substring-after.html:
* LayoutTests/fast/xpath/substring-nan-position-expected.txt:
* LayoutTests/fast/xpath/substring-nan-position.html:
* LayoutTests/fast/xpath/substring-non-positive-postion-expected.txt:
* LayoutTests/fast/xpath/substring-non-positive-postion.html:
* LayoutTests/fast/xpath/xpath-detached-iframe-resolver-crash-expected.txt:
* LayoutTests/fast/xpath/xpath-detached-iframe-resolver-crash.html:
* LayoutTests/fast/xpath/xpath-detached-import-assert-expected.txt:
* LayoutTests/fast/xpath/xpath-detached-import-assert.html:
* LayoutTests/fast/xpath/xpath-detached-nodes-expected.txt:
* LayoutTests/fast/xpath/xpath-detached-nodes.html:
* LayoutTests/fast/xpath/xpath-namespaces.html:
* LayoutTests/fast/xpath/xpath-non-ASCII-case-folding.html:
* LayoutTests/fast/xpath/xpath-template-element-expected.txt:
* LayoutTests/fast/xpath/xpath-template-element.html:

Canonical link: <a href="https://commits.webkit.org/292122@main">https://commits.webkit.org/292122@main</a>
</pre>
<!--EWS-Status-Bubble-Start-->
https://github.com/WebKit/WebKit/commit/507fcefafbcd92cfeccaccbef9458bdd4eeb7531

| Misc | iOS, visionOS, tvOS & watchOS  | macOS  | Linux |  Windows |
| ----- | ---------------------- | ------- |  ----- |  --------- |
| [✅ 🧪 style](https://ews-build.webkit.org/#/builders/38/builds/95011 "Passed style check") | [✅ 🛠 ios](https://ews-build.webkit.org/#/builders/131/builds/14606 "Built successfully") | [✅ 🛠 mac](https://ews-build.webkit.org/#/builders/138/builds/4463 "Built successfully") | [✅ 🛠 wpe](https://ews-build.webkit.org/#/builders/5/builds/100040 "Built successfully") | [✅ 🛠 win](https://ews-build.webkit.org/#/builders/59/builds/45509 "Built successfully") 
| | [✅ 🛠 ios-sim](https://ews-build.webkit.org/#/builders/130/builds/14895 "Built successfully") | [✅ 🛠 mac-AS-debug](https://ews-build.webkit.org/#/builders/123/builds/23028 "Built successfully") | [✅ 🧪 wpe-wk2](https://ews-build.webkit.org/#/builders/34/builds/72473 "Passed tests") | [✅ 🧪 win-tests](https://ews-build.webkit.org/#/builders/60/builds/29769 "Passed tests") 
| [✅ 🧪 webkitperl](https://ews-build.webkit.org/#/builders/11/builds/98012 "Passed tests") | [  ~~🧪 ios-wk2~~](https://ews-build.webkit.org/#/builders/132/builds/11120 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | [✅ 🧪 api-mac](https://ews-build.webkit.org/#/builders/18/builds/85793 "Passed tests") | [✅ 🧪 api-wpe](https://ews-build.webkit.org/#/builders/41/builds/52800 "Passed tests") | 
| | [✅ 🧪 ios-wk2-wpt](https://ews-build.webkit.org/#/builders/133/builds/10830 "Passed tests") | [✅ 🧪 mac-wk1](https://ews-build.webkit.org/#/builders/135/builds/3518 "Passed tests") | [✅ 🛠 wpe-cairo](https://ews-build.webkit.org/#/builders/65/builds/44848 "Built successfully") | 
| | [  ~~🧪 api-ios~~](https://ews-build.webkit.org/#/builders/13/builds/81024 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [✅ 🧪 mac-wk2](https://ews-build.webkit.org/#/builders/136/builds/3612 "Passed tests") | [✅ 🛠 gtk](https://ews-build.webkit.org/#/builders/2/builds/102085 "Built successfully") | 
| | [✅ 🛠 vision](https://ews-build.webkit.org/#/builders/128/builds/22053 "Built successfully") | [  ~~🧪 mac-AS-debug-wk2~~](https://ews-build.webkit.org/#/builders/122/builds/16094 "The change is no longer eligible for processing. Pull Request was already closed when EWS attempted to process it.") | [  ~~🧪 gtk-wk2~~](https://ews-build.webkit.org/#/builders/1/builds/81489 "The change is no longer eligible for processing. Commit was outdated when EWS attempted to process it.") | 
| | [✅ 🛠 vision-sim](https://ews-build.webkit.org/#/builders/121/builds/22300 "Built successfully") | [✅ 🧪 mac-wk2-stress](https://ews-build.webkit.org/#/builders/8/builds/81810 "Passed tests") | [✅ 🧪 api-gtk](https://ews-build.webkit.org/#/builders/21/builds/80876 "Passed tests") | 
| [✅ 🛠 🧪 merge](https://ews-build.webkit.org/#/builders/19/builds/20211 "Built successfully and passed tests") | [✅ 🧪 vision-wk2](https://ews-build.webkit.org/#/builders/126/builds/25430 "Passed tests") | [✅ 🧪 mac-intel-wk2](https://ews-build.webkit.org/#/builders/137/builds/2829 "Passed tests") | [✅ 🛠 playstation](https://ews-build.webkit.org/#/builders/134/builds/15295 "Built successfully") | 
| | [✅ 🛠 tv](https://ews-build.webkit.org/#/builders/127/builds/22026 "Built successfully") | | | 
| | [✅ 🛠 tv-sim](https://ews-build.webkit.org/#/builders/125/builds/21683 "Built successfully") | | | 
| | [✅ 🛠 watch](https://ews-build.webkit.org/#/builders/129/builds/25158 "Built successfully") | | | 
| | [✅ 🛠 watch-sim](https://ews-build.webkit.org/#/builders/124/builds/23424 "Built successfully") | | | 
<!--EWS-Status-Bubble-End-->

nt1m commented on LayoutTests/fast/xpath/null-namespace-in-html-expected.txt:
```diff
@@ -1,7 +1,6 @@
-This tests that XPath expressions with prefixes work correctly.
-
-PASS //div
+FAIL //div incorrect length (expected 1, actual 0)
```
This is now failing

zakariaridouh replied:
Weird, having a look. Probably using a function from `pre` that isn't there in the regular one.

aproskuryakov replied:
I didn't look deeply, but this looks liked a buggy test. Why is it using `document` and not `doc`?

zakariaridouh replied:
I think its failing because I removed the console element, which it directly relies on here:

```
checkSnapshot("//div", result, [document.getElementById("console")]);
```

aproskuryakov replied:
That's the immediate reason for results changing, but the root cause is that the test is buggy in the way I explained.

zakariaridouh replied:
Sure, I can have a look at fixing that test as a followup.

aproskuryakov replied:
Please don't collapse comments, especially those that are not fully addressed. I looked for comments, but didn't see this one before assuming that you just overlooked it.

aproskuryakov replied:
Fixing in a follow-up is OK.

zakariaridouh replied:
Sorry about that! Sounds good.

aproskuryakov commented on LayoutTests/fast/xpath/null-namespace-in-html-expected.txt:
```diff
@@ -1,7 +1,6 @@
 
 TEST COMPLETE
-
+This tests that XPath expressions with prefixes work correctly.
```
It's also not great that this moved to the bottom. You may want to use the `description()` function across the board for the harness to handle the title.

zakariaridouh committed:
[xpath] Modernize /fast/xpath tests
https://bugs.webkit.org/show_bug.cgi?id=289648
rdar://146892392

Reviewed by Tim Nguyen.

Modernize the tests in `fast/xpath` to use `test-js.js` and drop the
console element.

* LayoutTests/fast/xpath/4XPath/Borrowed/cz_20030217.html:
* LayoutTests/fast/xpath/4XPath/Borrowed/kd_20010423.html:
* LayoutTests/fast/xpath/4XPath/Borrowed/od_20000608.html:
* LayoutTests/fast/xpath/4XPath/Borrowed/rs_20010831.html:
* LayoutTests/fast/xpath/4XPath/Borrowed/sr_20021217.html:
* LayoutTests/fast/xpath/4XPath/Core/test_boolean_expr.html:
* LayoutTests/fast/xpath/4XPath/Core/test_core_functions-expected.txt:
* LayoutTests/fast/xpath/4XPath/Core/test_core_functions.html:
* LayoutTests/fast/xpath/4XPath/Core/test_literal_expr.html:
* LayoutTests/fast/xpath/4XPath/Core/test_location_path.html:
* LayoutTests/fast/xpath/4XPath/Core/test_node_test.html:
* LayoutTests/fast/xpath/4XPath/Core/test_nodeset_expr.html:
* LayoutTests/fast/xpath/4XPath/Core/test_numeric_expr.html:
* LayoutTests/fast/xpath/4XPath/Core/test_parser.html:
* LayoutTests/fast/xpath/4XPath/Core/test_predicate_list.html:
* LayoutTests/fast/xpath/4XPath/Core/test_step.html:
* LayoutTests/fast/xpath/ambiguous-operators-expected.txt:
* LayoutTests/fast/xpath/ambiguous-operators.html:
* LayoutTests/fast/xpath/attr-namespace.html:
* LayoutTests/fast/xpath/attribute-node-predicate.html:
* LayoutTests/fast/xpath/detached-subtree-invalidate-iterator.html:
* LayoutTests/fast/xpath/document-order.html:
* LayoutTests/fast/xpath/evaluate-twice.html:
* LayoutTests/fast/xpath/invalid-functions.html:
* LayoutTests/fast/xpath/node-name-case-sensitivity-expected.txt:
* LayoutTests/fast/xpath/node-name-case-sensitivity.html:
* LayoutTests/fast/xpath/null-namespace-in-html-expected.txt:
* LayoutTests/fast/xpath/null-namespace-in-html.html:
* LayoutTests/fast/xpath/position.html:
* LayoutTests/fast/xpath/py-dom-xpath/abbreviations.html:
* LayoutTests/fast/xpath/py-dom-xpath/data.html:
* LayoutTests/fast/xpath/py-dom-xpath/expressions.html:
* LayoutTests/fast/xpath/py-dom-xpath/functions-expected.txt:
* LayoutTests/fast/xpath/py-dom-xpath/functions.html:
* LayoutTests/fast/xpath/py-dom-xpath/nodetests.html:
* LayoutTests/fast/xpath/py-dom-xpath/paths.html:
* LayoutTests/fast/xpath/py-dom-xpath/predicates.html:
* LayoutTests/fast/xpath/reverse-axes.html:
* LayoutTests/fast/xpath/string-value.html:
* LayoutTests/fast/xpath/substring-after-expected.txt:
* LayoutTests/fast/xpath/substring-after.html:
* LayoutTests/fast/xpath/substring-nan-position-expected.txt:
* LayoutTests/fast/xpath/substring-nan-position.html:
* LayoutTests/fast/xpath/substring-non-positive-postion-expected.txt:
* LayoutTests/fast/xpath/substring-non-positive-postion.html:
* LayoutTests/fast/xpath/xpath-detached-iframe-resolver-crash-expected.txt:
* LayoutTests/fast/xpath/xpath-detached-iframe-resolver-crash.html:
* LayoutTests/fast/xpath/xpath-detached-import-assert-expected.txt:
* LayoutTests/fast/xpath/xpath-detached-import-assert.html:
* LayoutTests/fast/xpath/xpath-detached-nodes-expected.txt:
* LayoutTests/fast/xpath/xpath-detached-nodes.html:
* LayoutTests/fast/xpath/xpath-namespaces.html:
* LayoutTests/fast/xpath/xpath-non-ASCII-case-folding.html:
* LayoutTests/fast/xpath/xpath-template-element-expected.txt:
* LayoutTests/fast/xpath/xpath-template-element.html:

Canonical link: https://commits.webkit.org/292122@main
